### DISCUSSION TITLE
need help with the setup.

### DISCUSSION THREAD

**Discussion by danielnehrig** [mfussenegger/nvim-jdtls#151] [Q&A] [answered]
Lsp attaches just fine but when i open my project and go to any java file i get something like
`OrderController.java is a non-project file, only syntax errors are reported`

Project path `/home/dnehrig/code/work/order-service`
Project Structure `main contains source code of the app`
`tree -L 2`
```
├── Dockerfile
├── jdt.ls-java-project
│   ├── bin
│   └── src
├── Makefile
├── pom.xml
├── README.md
├── src
│   ├── main
│   └── test
```

config
`ftplugin/java.lua`
```lua
local lsp = require("plugins.lspconfig")
local capabilities = require("plugins.lspconfig").capabilities

local workspace_dir = vim.fn.fnamemodify(vim.fn.getcwd(), ":p:h:t")

local config = {
  capabilities = capabilities,
  flags = { debounce_text_changes = 500 },
  cmd = {
    "/usr/bin/java",
    "-Declipse.application=org.eclipse.jdt.ls.core.id1",
    "-Dosgi.bundles.defaultStartLevel=4",
    "-Declipse.product=org.eclipse.jdt.ls.core.product",
    "-noverify",
    "-Xms1G",
    "-jar",
    "/usr/share/java/jdtls/plugins/org.eclipse.equinox.launcher_1.6.300.v20210813-1054.jar",
    "-configuration",
    "/home/dnehrig/config_linux",
    "-data",
    "/home/dnehrig/code/work/" .. workspace_dir,
  },

  root_dir = require("jdtls.setup").find_root({ ".git", "pom.xml" }),
  on_attach = function(client, bufnr)
    lsp:on_attach(client, bufnr)
  end,
}
require("jdtls").start_or_attach(config)
```

**Comment by danielnehrig** [comment]
to clarify LSP attaches just fine and runs i get completion and diagnostics for that one buffer
but it seems like it does not index the workspace

**Comment by danielnehrig** [comment]
ok funny enough everything works if i try it out on a fresh vm setup
i've tried almost everything i deleted every java,jre,jdk,maven,gradle package and reinstalled everything
i can not get it to work might there be some jdtls cache folder that im missing that i need to clean ?

**Reply by mfussenegger** [reply]
The workspace folder that you point to with `-data` contains project specific information. Other than that there is also a `.project` file within each project directory.

**Reply by danielnehrig** [reply]
i've understood it like that:
-data needs to point to the location of the project where pom or your repo is

i've fixed my issue by doing something like that `-data ~/code/workspaces/projectname` instead of what i had before where my git repo was located

**Reply by mfussenegger** [reply]
No it shouldn't point to the project but to a dedicated workspace folder. (That's an eclipse thing).

**Comment by danielnehrig** [comment, marked as answer]
i fixed it by not pointing the `-data` path to the exact same folder as the repository name but instead to the parent so
project: ~/code/java/my-app
-data: ~/code/java/workspaces/my-app
